walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,64 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      //Return the times that a meeting can happen today
+      //MeetingRequest holds all the information regarding atendees, and length of meeting
+      //Event has a set of attendees and a specific timerange the atendees are meeting 
+      //TimeRange is a specific set of time a meeting can start and end
+    Collection<TimeRange> slots = new HashSet<TimeRange>();
+    if (request.getDuration() < TimeRange.WHOLE_DAY.duration()) {
+        slots.add(TimeRange.WHOLE_DAY); //Initializing base case
+    }
+
+    for (Event event : events) {
+        for (String attendee : event.getAttendees()) {
```

Comment:
I'm wondering if there's a way we don't have to iterate over both of these is a nested manner. I don't see one right now but I'll think about it.